docs/backends/macos.rst

```diff
@@ -28,7 +28,31 @@ As stated above, this will however only work the macOS machine that performed
 the scan and thus cached the device as ``243E23AE-4A99-406C-B317-18F1BD7B4CBE``.
 
 There is also no pairing functionality implemented in macOS right now, since it does not seem
-to be any explicit pairing methods in the COre Bluetooth.
+to be any explicit pairing methods in the Core Bluetooth.
+
+CoreBluetooth does not differentiate between data from a notification, and data from a read.
+This can cause confusion in cases where a device may send a notification message on a characteristic
+that is an indication that the characteristic needs to be read again.
+
+Bleak has can accept a ``notification_discriminator`` callback in the ``cb`` dict parameter that is
+passed to the start_notify function that can differentiate between these types of data.
+
+.. code-block:: python
+
+    async def reread(char):
+        data = await client.read_gatt_char(char)
+
+    async def notification_handler(char, data):
+        asyncio.create_task(reread(char))
+
+    def notification_check_handler(data):
+        return len(data) == 1
```

Comment:
```suggestion
        # We can identify notifications on this characteristic because they
        # only contain 1 byte of data. Read responses will have more than
        # 1 byte.
        return len(data) == 1
```

Reply:
Comments added, documentation examples cleaned up to not be buggy.